### ISSUE TITLE
Attribute "ports" on Container Does Not Include Ports

### ISSUE THREAD

**Issue by ClaytonNorthey92** [docker/docker-py#2681]
I am running a (postgres) container using the Docker Python SDK.

I am trying to bind a random host port to port `5432` in the container.  According to the [documentation](https://docker-py.readthedocs.io/en/stable/containers.html#):
> None, to assign a random host port. For example, {'2222/tcp': None}.

This is my setup:
```python
import docker


if __name__ == '__main__':
    client = docker.from_env()
    container = client.containers.run('postgres:9.6', None,
        detach=True,
        ports= {
            '5432/tcp': None # bind to a random port
        },
        network_mode='bridge',
        environment={
            'POSTGRES_PASSWORD': 'password'
        },
    )

    print(container.attrs['NetworkSettings'])
    print(container.ports)
```

The output from the `print` statements:
```shell
{'Bridge': '', 'SandboxID': '', 'HairpinMode': False, 'LinkLocalIPv6Address': '', 'LinkLocalIPv6PrefixLen': 0, 'Ports': {}, 'SandboxKey': '', 'SecondaryIPAddresses': None, 'SecondaryIPv6Addresses': None, 'EndpointID': '', 'Gateway': '', 'GlobalIPv6Address': '', 'GlobalIPv6PrefixLen': 0, 'IPAddress': '', 'IPPrefixLen': 0, 'IPv6Gateway': '', 'MacAddress': '', 'Networks': {'bridge': {'IPAMConfig': None, 'Links': None, 'Aliases': None, 'NetworkID': '', 'EndpointID': '', 'Gateway': '', 'IPAddress': '', 'IPPrefixLen': 0, 'IPv6Gateway': '', 'GlobalIPv6Address': '', 'GlobalIPv6PrefixLen': 0, 'MacAddress': '', 'DriverOpts': None}}}
{}
```

According to [this pull request](https://github.com/docker/docker-py/pull/1882) the `.ports` property should have the mapped port.

Python version: 
```
Python 3.7.9
```

Docker SDK version: 
```
Name: docker
Version: 4.3.1
``` 

And I can verify that the container indeed does have a bound port:
```
$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                     NAMES
e97a2bb377fb        postgres:9.6        "docker-entrypoint.s…"   3 seconds ago       Up 3 seconds        0.0.0.0:32780->5432/tcp   stupefied_hellman
```

**Comment by steffenhk** [comment]
I had the same problem. Adding a container.reload() after calling run() seems to refresh the ports dict.

**Comment by ClaytonNorthey92** [comment]
I think the issue I was seeing has to do with the container's state after you create the container via the docker API.

I used @steffenhk 's advice and used `container.reload()` like so:
```python
print(container.attrs['State'])
container.reload()
print(container.attrs['State'])
```

and got the output...
```python
{'Status': 'created', 'Running': False, 'Paused': False, 'Restarting': False, 'OOMKilled': False, 'Dead': False, 'Pid': 0, 'ExitCode': 0, 'Error': '', 'StartedAt': '0001-01-01T00:00:00Z', 'FinishedAt': '0001-01-01T00:00:00Z'}

{'Status': 'running', 'Running': True, 'Paused': False, 'Restarting': False, 'OOMKilled': False, 'Dead': False, 'Pid': 14020, 'ExitCode': 0, 'Error': '', 'StartedAt': '2020-12-14T21:27:13.0983733Z', 'FinishedAt': '0001-01-01T00:00:00Z', 'Health': {'Status': 'starting', 'FailingStreak': 0, 'Log': []}}
```

One option would be to add a healthcheck, but even then you would have to [poll the health status like this function does](https://github.com/docker/docker-py/blob/845d31dbd280f6c28cdb8594c6ff9d9f4b038bea/tests/integration/api_healthcheck_test.py#L7)

tldr; this isn't an issue with the `docker-py` code, this is me not handling starting the postgres container correctly.  I would need a separate way to determine the container was up and running (such as a healthcheck), then `reload()` and check the port number
